src/EditorFeatures/Test/Diagnostics/DiagnosticProviderTestUtilities.cs

```diff
@@ -49,7 +49,7 @@ private static IEnumerable<Diagnostic> GetDiagnostics(DiagnosticAnalyzer analyze
                     syntaxNodeAnalyzerService: nodeInBodyAnalyzerService,
                     hostDiagnosticUpdateSource: exceptionDiagnosticsSource,
                     cancellationToken: CancellationToken.None, 
-                    testOnly_DonotCatchAnalyzerExceptions: donotCatchAnalyzerExceptions);
+                    testOnly_RethrowAnalyzerException: donotCatchAnalyzerExceptions);
```

Comment:
No.  The exception should never make it to the compiler.  Even in test code.  

We should remove this.  Have the code log the exception instead and assert it was thrown afterwards.